### PR TITLE
Check coinbase inputs

### PR DESCRIPTION
**Pull request by farsider350**
Implement auxpow POW checks to combat the malformed auxpow coinbase
when there's empty array of inputs causing segmentation fault.

Dingocoin versions affected: All prior releases

Who is impacted? Potentially all dingocoind node operators (miners, exchanges, businesses).

Impact if exploited: Potential to remotely crash dingocoind and halt the Dingocoin network. User funds are NOT at risk.

Details:
Commit https://github.com/dingocoin/dingocoin/commit/bc8cca48968dfa3f60b5eae6a2b92bdd2870eee3 introduced AuxPow support into dogecoind/dingocoind. AuxPow is a feature that allows Dingocoin and other blockchain networks to share proof of work. While the AuxPow data is not validated in entirety (since
dingocoind has no way to know the consensus rules of other blockchains), there is verification of the proof of work of the
AuxPow payload. This is done in AuxPow::check() in src/auxpow.cpp. The vulnerability in question relates to this line:

const CScript script = tx->vin[0].scriptSig;

Note the assumption that tx->vin is not empty. No validation is done to ensure tx->vin contains one or more entries.

### CODE REVIEW

File: src/validation.cpp [hunk 1]
```diff
@@ -2911,11 +2911,10 @@ bool CheckBlockHeader(const CBlockHeader& block, CValidationState& state, bool f
 
 
         // Can't do these since current aux chain id is not known.
```
Comment (seanPhill) [inline src/validation.cpp:2913]: Shouldn't line 2913 be deleted along with the following five comment lines? Line 2913 must, I suppose be referring to the already commented out lines from 2914 to 2918.

Comment (farsider350) [reply src/validation.cpp:2913]: Yes

**Review by seanPhill — APPROVED** [review]
Looks good.

File: src/validation.cpp [hunk 2]
```diff
@@ -2909,13 +2909,10 @@ bool CheckBlockHeader(const CBlockHeader& block, CValidationState& state, bool f
...
-        // Can't do these since current aux chain id is not known.
-        //if (!block.auxpow->check(block.GetHash(), block.GetChainId(), params))
-        //    return error("%s : AUX POW is not valid", __func__);
-        //if (!CheckProofOfWork(block.auxpow->getParentBlockPoWHash(), block.nBits, params))
-        //    return error("%s : AUX proof of work failed", __func__);
-        //
+        if (!CheckProofOfWork(block.auxpow->getParentBlockPoWHash(), block.nBits, params))
+            return error("%s : AUX proof of work failed", __func__);
+        if (!block.auxpow->check(block.GetHash(), block.GetChainId(), params))
+            return error("%s : AUX POW is not valid", __func__);
```
Comment (Twinky-kms) [inline src/validation.cpp:2915]: I think we should save this for a different PR given the nature of the intended fix.

**Review by Twinky-kms — APPROVED** [review]
lgtm assuming CD passes.

**Pull request merged**